tests/Db/OldPasswordMapperTest.php

```diff
@@ -57,46 +59,110 @@ protected function tearDown() {
-
-		$oldPassword = new OldPassword();
-		$oldPassword->setUid($this->testUID);
-		$oldPassword->setPassword('testpass2');
-		$oldPassword->setChangeTime(\OC::$server->getTimeFactory()->getTime()+1);
-		$this->mapper->insert($oldPassword);
-
-		$oldPassword = new OldPassword();
-		$oldPassword->setUid($this->testUID);
-		$oldPassword->setPassword('testpass3');
-		$oldPassword->setChangeTime(\OC::$server->getTimeFactory()->getTime()+2);
-		$this->mapper->insert($oldPassword);
+		foreach ($this->testUIDs as $index => $uid) {
+			$oldPassword = new OldPassword();
+			$oldPassword->setUid($uid);
+			$oldPassword->setPassword("{$uid}testpass1");
+			$oldPassword->setChangeTime($baseTime);
+			$this->mapper->insert($oldPassword);
+
+			$oldPassword = new OldPassword();
+			$oldPassword->setUid($uid);
+			$oldPassword->setPassword("{$uid}testpass2");
+			$oldPassword->setChangeTime($baseTime + (10 * ($index + 1)));
+			$this->mapper->insert($oldPassword);
+
+			$oldPassword = new OldPassword();
+			$oldPassword->setUid($uid);
+			$oldPassword->setPassword("{$uid}testpass3");
+			$oldPassword->setChangeTime($baseTime + (20 * ($index + 1)));
+			$this->mapper->insert($oldPassword);
+		}
 	}
 
 	public function testGetOldPasswords() {
-		$this->assertCount(0, $this->mapper->getOldPasswords($this->testUID,3));
-		$this->addInitialTestEntries();
+		$uid = $this->testUIDs[0];
+		$this->assertCount(0, $this->mapper->getOldPasswords($uid, 3));
+		$this->addInitialTestEntries(\time());
 
-		$oldPasswords = $this->mapper->getOldPasswords($this->testUID,2);
+		$oldPasswords = $this->mapper->getOldPasswords($uid, 2);
 
 		$this->assertCount(2, $oldPasswords);
-		$this->assertNotSame("testpass1", $oldPasswords[0]->getPassword());
-		$this->assertNotSame("testpass1", $oldPasswords[1]->getPassword());
+		$this->assertNotSame("{$uid}testpass1", $oldPasswords[0]->getPassword());
+		$this->assertNotSame("{$uid}testpass1", $oldPasswords[1]->getPassword());
 
-		$this->assertCount(3, $this->mapper->getOldPasswords($this->testUID,3));
+		$this->assertCount(3, $this->mapper->getOldPasswords($uid, 3));
 	}
 
 	public function testLatestPassword() {
-		$this->assertNull($this->mapper->getLatestPassword($this->testUID));
-		$this->addInitialTestEntries();
+		$uid = $this->testUIDs[0];
+		$this->assertNull($this->mapper->getLatestPassword($uid));
+		$this->addInitialTestEntries(\time());
+
+		$latestPassword = $this->mapper->getLatestPassword($uid);
+
+		$this->assertSame("{$uid}testpass3", $latestPassword->getPassword());
+	}
+
+	public function testGetPasswordsAboutToExpireAllOk() {
+		$baseTime = \time();
+		$this->addInitialTestEntries($baseTime);
+
+
+		$passwordList = $this->mapper->getPasswordsAboutToExpire($baseTime + 14);
+		// last password change after the timestamp, so we shouldn't get any result
+		$this->assertCount(0, $passwordList);
+	}
+
+	public function testGetPasswordsAboutToExpireSomePassMarked() {
+		$baseTime = \time();
```

Comment:
maybe use a hard-coded value for deterministic tests ?

Reply:
I don't think it's needed. The rest of the timestamps are based on whatever the `$baseTime` is, and we have control over the current time via the timeFactory (also based on the `$baseTime`). There won't be problems about this test failing randomly